src/Discord/Internal/Gateway/EventLoop.hs

```diff
@@ -83,12 +90,15 @@ connectionLoop auth intent gatewayHandle log = outerloop LoopStart
     -- | Main connection loop. Catch exceptions and reconnect.
     outerloop :: LoopState -> IO ()
     outerloop state = do
+        -- Read the gateway address, since each new connection to the gateway
+        -- may have a different Host to connect to.
```

Comment:
I think this comment is unnecesary, because the code is straightforward and you've added good docs on `gatewayHandleHostname` above